bonsai-core/src/main/scala/com/stripe/bonsai/IndexedBitSet.scala

```diff
@@ -0,0 +1,306 @@
+package com.stripe.bonsai
+
+import scala.annotation.switch
+import scala.collection.immutable.BitSet
+import scala.collection.mutable.{ Builder, ArrayBuilder }
+
+/**
+ * IndexedBitSet is an immutable set for storing non-negative integer
+ * values.
+ * 
+ * This data structure stores non-negative integers. It has O(1)
+ * lookup and rank operations. It also supports O(polylog n) select
+ * operations, which are not currently implemented.
+ *
+ * To support fast rank, this bitset requires ~1.37n bits, rather than
+ * the n bits that other bitsets (e.g. `scala.BitSet`) would use.
+ *
+ * Unlike some immutable structures, there is no form of structural
+ * sharing here. Thus, the only way to "modify" one of these
+ * structures is to build a new one that has the desired modifications
+ * (which will not be efficient if done frequently).
+ */
+final class IndexedBitSet(
```

Comment:
It's hard for me to tell, since github doesn't put renamed files side-by-side, but aside from toSeq/toString, did anything really change here, implementation wise?